scipy/special/cephes/hyp2f1.c

```diff
@@ -83,6 +83,43 @@ static double hyt2f1(double a, double b, double c, double x, double *loss);
 static double hys2f1(double a, double b, double c, double x, double *loss);
 static double hyp2f1ra(double a, double b, double c, double x,
 		       double *loss);
+static double pochhammer(double x, double n);
+static double factorial(double k);
+static double hyp2f1_indeterminate(double a, double b, double x);
+
+/*
+    See http://mathworld.wolfram.com/PochhammerSymbol.html
+*/
+static double pochhammer(double x, double n)
+{
+    return gamma(x + n) / gamma(x);	
+}
+
+/*
+    See http://mathworld.wolfram.com/GammaFunction.html
+*/
+static double factorial(double k)
+{
+    return gamma(k + 1);
+}
+
+/*
+    See https://math.stackexchange.com/questions/2457831/hypergeometric-expansion-approximation
+*/
+static double hyp2f1_indeterminate(double a, double b, double x)
```

Comment:
Move all helper functions after the main function, like the other special cases.

Also, more specific function name could be better (regardless of the fact that the other functions follow fortran 66 naming style).

Please also prefer citing an appropriate DLMF or abramowitz&stegun formula over stackoverflow posts.

Reply:
Give me a name for the function. I'm terrible at names.